src/libraries/Native/Unix/System.Globalization.Native/CMakeLists.txt

```diff
@@ -28,7 +28,7 @@ if(CLR_CMAKE_TARGET_UNIX)
                 message(FATAL_ERROR "Cannot find libicucore, skipping build for System.Globalization.Native. .NET globalization is not expected to function.")
                 return()
             endif()
-            add_definitions(-DOSX_ICU_LIBRARY_PATH=\"${ICUCORE}\")
+            add_definitions(-DOSX_ICU_LIBRARY_PATH=\"/usr/lib/libicucore.dylib\")
```

Comment:
nit: does setting `CMAKE_FIND_LIBRARY_SUFFIXES=.dylib` before the introspection line above (find_library) and unsetting it after the introspection works? That will save us using the hard-coded path.

Reply:
e.g. https://github.com/dotnet/runtime/blob/827a523bfc7fd6c3835e62f4fb5fdacafd0e0060/src/libraries/Native/Unix/System.Security.Cryptography.Native/CMakeLists.txt#L11-L13

Reply:
let's do that in a follow-up PR to unblock CI.

Reply:
it might, but I see no reason why we should not use the hardcoded path since that is where ICU will always be on macOS.

Reply:
That defeats the purpose of using `find_library` (the call above). :)

Reply:
yeah I think we can remove that ;)

Reply:
actually, it is used here so we can't remove it: https://github.com/dotnet/runtime/blob/56c882ecbbac7113a09321b013809f9de27b9eec/src/libraries/Native/Unix/System.Globalization.Native/configure.cmake#L17

Reply:
Is CMAKE_REQUIRED_LIBRARIES setting to libicucore.tbd considered ok? maybe we need to change that as well; .tbd is a metadata (YAML) file.

Even better, if we could avoid introducing more hardcoded paths using `set(CMAKE_FIND_LIBRARY_SUFFIXES .dylib)`, it will at least keep it consistent with other usages of `find_*` macros (under platform conditions or platform specific cmake files), IMHO. e.g. in theory, someone can build libicu locally on macOS (for experiments) and have it installed in other place present in PATH.

Reply:
doesn't seem to be needed though, will send a PR to remove it ;)

Reply:
Perhaps this is what you're discussing here, but from https://gitlab.kitware.com/cmake/cmake/-/issues/21007

> The .tbd files in the SDK are the preferred way to link to system libraries on macOS.  This is in preparation for macOS 11 where the /usr/lib/*.dylib files do not even exist on disk.

Is it correct that `/usr/lib/libicucore.dylib` does not exist on Big Sur? If so will this change break there?

cc @ViktorHofer

Reply:
If this doesn't exist in Big Sur I will hit it soon as I will be fixing: https://github.com/dotnet/runtime/issues/39583

So if that's the case I will also fix this to work on Big Sur.

Reply:
@danmosemsft @safern I commented on the cmake gitlab issue, but copying here:

I checked on a macOS 11 system and the file is indeed not on disk but you can apparently still load it via `dlopen()` so we should still be able to use it.

They probably do some tricks in the dynamic loader to allow loading libicucore.dylib even though it doesn't exist at that location on disk. We don't know whether Apple will keep that or remove at some later point though.